scene/resources/shape_2d.cpp

```diff
@@ -105,6 +109,15 @@ void Shape2D::_bind_methods() {
 	ADD_PROPERTY(PropertyInfo(Variant::FLOAT, "custom_solver_bias", PROPERTY_HINT_RANGE, "0,1,0.001"), "set_custom_solver_bias", "get_custom_solver_bias");
 }
 
+bool Shape2D::is_collision_outline_enabled() {
+#ifdef TOOLS_ENABLED
+	if (Engine::get_singleton()->is_editor_hint()) {
+		return true;
+	}
+#endif
+	return GLOBAL_DEF("debug/shapes/collision/draw_2d_outlines", true);
+}
```

Comment:
Isn't #46410 also a performance issue when editing a huge tilemap in the editor?

Reply:
I guess not because of #19858, but it might become an issue if #37212 or some form of it is merged.

Reply:
It should be fine in any case. I was testing with `--debug-collisions` and from what I've seen, the editor doesn't redraw on every frame so the performance issues don't occur.